packaging/greenboot/functions.sh

```diff
@@ -142,6 +142,38 @@ function namespace_images_downloaded() {
     return 0
 }
 
+# Check if all the deployments in a given namespace are ready.
+#
+# args: None
+# env1: 'CHECK_DEPLOY_NS' environment variable for the namespace to check
+# env2: 'CHECK_DEPLOY_CT' environment variable for the deployment count to check
+# return: 0 if deployment is ready, or 1 otherwise
+function namespace_deployment_ready() {
+    local -r ns="${CHECK_DEPLOY_NS}"
+    local -r ct="${CHECK_DEPLOY_CT}"
+
+    local -r output=$(${OCGET_CMD} deployments ${OCGET_OPT} \
+        -n "${ns}" \
+        -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.spec.replicas}{" "}{.status.readyReplicas}{"\n"}{end}' \
+        2>/dev/null)
+
+    # Count non-empty lines in the output and compare with the
+    # number of expected deployments
+    local -r lcount=$(echo -e "${output}" | grep -vc '^$')
+    if [ "${lcount}" -ne "${ct}" ] ; then
+        return 1
+    fi
+
+    # Count the number of available replicas that is not equivalent
+    # to the number of ready replicas
+    local -r rcount=$(echo -e "${output}" | awk '$2 != $3 {print $1}' | grep -vc '^$')
+    if [ "${rcount}" -ne 0 ] ; then
+        return 1
+    fi
```

Comment:
Instead of manually comparing ready and expected replicas we could use on of these:
```
$ oc rollout status deployment -n openshift-operator-lifecycle-manager        
Waiting for deployment "catalog-operator" rollout to finish: 0 of 1 updated replicas are available...
deployment "catalog-operator" successfully rolled out
deployment "olm-operator" successfully rolled out
```

```
$ oc wait deployment -n openshift-operator-lifecycle-manager catalog-operator --for=condition=Available --timeout=300s
deployment.apps/catalog-operator condition met
```

Reply:
Fixed.